codegen-test/model/misc.smithy

```diff
@@ -108,3 +112,67 @@ union AUnion {
     string: String,
     time: Timestamp,
 }
+
+/// This operation tests that the response code defaults to @http's code
+@http(method: "GET", uri: "/responseCodeDefault", code: 418)
+@httpRequestTests([
+    {
+        id: "ResponseCodeDefaultTest",
+        protocol: "aws.protocols#restJson1",
+        uri: "/responseCodeDefault",
+        method: "GET",
+    }
+])
+@httpResponseTests([
+    {
+        id: "ResponseCodeDefaultTest",
+        protocol: "aws.protocols#restJson1",
+        code: 418,
+    }
+])
+operation ResponseCodeDefault {
+    input: ResponseCodeDefaultInput,
+    output: ResponseCodeDefaultOutput,
+}
+
+@input
+structure ResponseCodeDefaultInput {}
+
+@output
+structure ResponseCodeDefaultOutput {}
+
+/// This operation tests that @httpResponseCode is @required
+/// and is used over @http's code
+@http(method: "GET", uri: "/responseCodeRequired", code: 200)
+@httpRequestTests([
+    {
+        id: "ResponseCodeRequiredTest",
+        protocol: "aws.protocols#restJson1",
+        uri: "/responseCodeRequired",
+        method: "GET",
+    }
+])
+@httpResponseTests([
+    {
+        id: "ResponseCodeRequiredTest",
+        protocol: "aws.protocols#restJson1",
+        code: 418,
+        params: {"responseCode": 418}
+    }
+])
+operation ResponseCodeRequired {
+    input: ResponseCodeRequiredInput,
+    output: ResponseCodeRequiredOutput,
+}
+
+@input
+structure ResponseCodeRequiredInput {}
+
+@output
+structure ResponseCodeRequiredOutput {
+    @required
+    @httpResponseCode
+    responseCode: HttpResponseCode,
```

Comment:
```suggestion
    responseCode: Integer,
```

You can just use the shape in the prelude to not have to define `HttpResponseCode`.

Reply:
https://awslabs.github.io/smithy/1.0/spec/core/prelude-model.html